hedera-mirror-rosetta/app/domain/types/transaction.go

```diff
@@ -11,16 +11,20 @@ type Transaction struct {
 }
 
 // FromRosettaTransaction populates domain type Transaction from Rosetta type Transaction
-func (t *Transaction) FromRosettaTransaction(rTransaction *rTypes.Transaction) {
-	t.ID = rTransaction.TransactionIdentifier.Hash
-
+func FromRosettaTransaction(rTransaction *rTypes.Transaction) (*Transaction, error) {
 	operations := make([]*Operation, len(rTransaction.Operations))
 	for i, rosettaO := range rTransaction.Operations {
-		o := &Operation{}
-		o.FromRosettaOperation(rosettaO)
+		o, err := FromRosettaOperation(rosettaO)
+		if err != nil {
+			return nil, err
```

Comment:
I think you should add a new Error to [errors](https://github.com/LimeChain/hedera-mirror-node/blob/master/hedera-mirror-rosetta/app/errors/errors.go) and return it.